### PR TITLE
[WIP] light: Refactor SyslogNgConfig API (external/internal)

### PR DESCRIPTION
**Pull request by mitzkia**
This patch set provides a general API for syslog-ng config drivers with the following changes:
- `working_dir` class variable has been converted into a global variable (it was also a main dependency of SyslogNgConfig classes)
- After that driver/parser/filter related wrapper functions (such as create_file_source()) could be removed from SyslogNgConfig() (they were boilerplates).
- Start using own classes (such as FileSource, FileDestination) for creating driver/parser/filter like objects in testcases
- Change functionality of SourceDriver and DestinationDriver to SourceWriter and DestinationReader, 
from now these classes are responsible only for general driver related input/output functionalities.
- Expose file related option handling into a new class: `FileBasedOptionHandler()`
- Follow the changes in functional testcases

Todos:
- find a better place to config.stringify()
- separate logpath related wrappers from SyslogNgConfig
- automatically create logpath (if needed)
- add missing functional testcase: change file-path during testrun
- add missing functional testcase: add new file based option during testrun (an example usage for OptionHandler() class)

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: tests/python_functional/src/syslog_ng_config/statements/sources/source_writer.py [hunk 1]
```diff
@@ -24,22 +24,20 @@
...
+        self.driver_io = None
 
-    def __init__(self, IOClass, positional_parameters=[], options={}):
-        self.__IOClass = IOClass
-        self.__writer = None
-        self.positional_parameters = positional_parameters
-        self.options = options
+    def construct_writer(self, driver_resource):
+        if not self.driver_io or (self.actual_driver_resource != driver_resource):
+            self.actual_driver_resource = driver_resource
```
Comment (furiel) [inline tests/python_functional/src/syslog_ng_config/statements/sources/source_writer.py]: What is the purpose of `actual_driver_resource`?

File: tests/python_functional/src/syslog_ng_config/statements/sources/source_writer.py [hunk 2]
```diff
@@ -24,22 +24,20 @@
...
+        self.driver_io_cls = driver_io_cls
+        self.actual_driver_resource = None
+        self.driver_io = None
 
-    def __init__(self, IOClass, positional_parameters=[], options={}):
-        self.__IOClass = IOClass
-        self.__writer = None
-        self.positional_parameters = positional_parameters
-        self.options = options
+    def construct_writer(self, driver_resource):
```
Comment (furiel) [inline tests/python_functional/src/syslog_ng_config/statements/sources/source_writer.py]: This should be called constructor_parameters, and should be handled as a list. Or better, this method should be pass, and overridden by a FileSourceWriter.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by mitzkia** [comment]
Closing now. This was not a complete concept. 
New PR is #2804

**Pull request closed without merging**